### PR TITLE
Added span context to exchanged in headers

### PR DESCRIPTION
**Pull request by Kunal-Jha**
Added span context to exchanged in headers
Zalando ticket: ARUHA-2467

Description
This PR  limits tracing only to the validation layer and publishing to Kafka and allows the passing of trace context in headers. 

https://jira.zalando.net/browse/ARUHA-2547

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/filters/LoggingFilter.java [hunk 1]
```diff
@@ -138,14 +160,18 @@ protected void doFilterInternal(final HttpServletRequest request,
...
         } finally {
             if (!request.isAsyncStarted()) {
-                final RequestLogInfo requestLogInfo = new RequestLogInfo(request, start);
+                final RequestLogInfo requestLogInfo = new RequestLogInfo(request, startTime);
                 logRequest(requestLogInfo, response.getStatus(), publishingSpan);
             }
+            final Map<String, String> spanContextToInject = new HashMap<>();
+            GlobalTracer.get().inject(publishingSpan.context(),
+                    HTTP_HEADERS, new TextMapInjectAdapter(spanContextToInject));
+            response.setHeader("span_ctx",spanContextToInject.toString());
```
Comment (rcillo) [inline src/main/java/org/zalando/nakadi/filters/LoggingFilter.java]: It would be nice to move `span_ctx` to a constant. Also add space after `,`.

**Comment by Kunal-Jha** [comment]
@rcillo  Added the docs in 4bd52a5

**Comment by rcillo** [comment]
👍

**Comment by Kunal-Jha** [comment]
👍

**Pull request merged**